proposals/yaml_signing_and_verification.md

```diff
@@ -0,0 +1,253 @@
+
+Here is an example of signing an nginx pod using a static key:
+
+```
+ kubectl sigstore sign -f nginx-pod.yaml  --key cosign.key
+```
+
+This produces a `nginx-pod.yaml.signed` declaration that looks like this:
+
+```
+ apiVersion: v1
+kind: Pod
+metadata:
+  annotations:
+    cosign.sigstore.dev/message: H4sIAAAAAAAA/wAzAcz+H4sIAAAAAAAA/+ySPU/DMBCGM+dXnLqH2v2IW6/sKANiP5IjMjg+y3YLBfHfUdwKmEB0qZDyLCf7PT/+kOdp8POWBx8oRuP6KmGo+tdlLWW9larezl1v3Evlubs64GCLMxBCCLVej1WqtfheM0u1KuRK1lJJpepFIRZCrZYFiHM2+yu7mDAUQjya4ce+3/LTXT7rPwG9uaMQDTsNe1k+GddpaLgrB0rYYUJdAqBznDAZdlHD23sJYPGebBwzALSWn6uOLCWKGmYPaCPNSgCHA2nw3FX5D5XRUzsuadklNI5CFlRgBuxJQ27SFhPFlMV5vtlZ27A17UHDDe0p5OioPmrHseeQTsepvvwNh6RhIzYiJwA+cOKWrYbb66a89NtPTExMXJKPAAAA//+lixZxAAgAAAEAAP//QYqwEDMBAAA=
+    cosign.sigstore.dev/signature: MEQCIGJ9cjguB6TXcMGCDyWyCRyVOb6YOrHDo6S4VtZGuvVjAiA1qm5N/koj94I4JtL5BywhuBVxS7VpEkGLtcwIkVDogg==
+  labels:
+    allow-deletes: "false"
+  name: pod-nginx
+spec:
+  containers:
+  - image: nginx:latest
+    imagePullPolicy: Never
+    name: nginx
+    ports:
+    - containerPort: 8080
+      protocol: TCP
+
+```
+
+This declaration can then be verified by a Kyverno policy rule. This policy checks that all pods are signed using a static key:
+
+```
+---
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: check-signatures
+spec:
+  rules:
+    - name: check-manifest
+      match:
+        resources:
+          kinds:
+            - Pod
+      validate:
+        manifests:
+          - key: |-
+            -----BEGIN PUBLIC KEY-----
+                      ...
+            -----END PUBLIC KEY-----
+```
+
+This option of signing and verifying YAMLs is compatible with GitOps workflows and does not require an OCI registry to store signed resource definitions.
+
+## Implementation
+
+Although referred to as “YAML signing” this feature may be more accurately described as verifying the integrity of a Kubernetes manifest or resource declaration. YAML is used as a canonical format during signing and verification.
+
+Since Kubernetes resources are typically mutated during admission controls by native and 3rd party controllers, this feature needs to be able to allow certain mutations, and ignore them during serialization and verification. 
+
+There are two approaches to handle the defaults and other mutations:
+
+1. Allow users to specify an `ignoreFields` in the policy: this is useful when there are few changes in the resource manifest, and and most changes are isolated to the `status` element. The drawback of this approach is that each field needs to be maintained in the policy.
+
+2. Use the API sever `dry-run` feature: this is useful when there are several changes made for defaults and by controllers to the resource. For example, a pod has several fields across its `spec` and `status` elements that get modified over its lifecycle. The drawback of this approach is that Kyverno needs permissions to create the resource via a `dry-run`. 
+
+## Requirements
```

Comment:
Another requirement must be that support for this is brought to the Kyverno CLI and not just the webhook.